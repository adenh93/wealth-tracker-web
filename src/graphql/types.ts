export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type CryptoCurrency = {
  __typename?: 'CryptoCurrency';
  id: Scalars['Int'];
  name: Scalars['String'];
  rank: Scalars['Int'];
  symbol: Scalars['String'];
};

export type CryptoCurrencyHolding = {
  __typename?: 'CryptoCurrencyHolding';
  cryptoCurrency?: Maybe<CryptoCurrency>;
  cryptoCurrencyId: Scalars['Int'];
  holdings: Scalars['Float'];
  id: Scalars['Int'];
  netHoldingsValue: Scalars['Float'];
  percentChange24h: Scalars['Float'];
  price: Scalars['Float'];
};

export type CryptoCurrencyHoldingsAddInput = {
  cryptoCurrencyId: Scalars['Int'];
  holdings: Scalars['Float'];
};

export type CryptoCurrencyHoldingsUpdateInput = {
  holdings: Scalars['Float'];
  id: Scalars['Int'];
};

export type Mutation = {
  __typename?: 'Mutation';
  cryptoCurrencyHoldingsAdd?: Maybe<CryptoCurrencyHolding>;
  cryptoCurrencyHoldingsDelete?: Maybe<CryptoCurrencyHolding>;
  cryptoCurrencyHoldingsUpdate?: Maybe<CryptoCurrencyHolding>;
};


export type MutationCryptoCurrencyHoldingsAddArgs = {
  input: CryptoCurrencyHoldingsAddInput;
};


export type MutationCryptoCurrencyHoldingsDeleteArgs = {
  id: Scalars['Int'];
};


export type MutationCryptoCurrencyHoldingsUpdateArgs = {
  input: CryptoCurrencyHoldingsUpdateInput;
};

export type Query = {
  __typename?: 'Query';
  cryptoCurrencies?: Maybe<Array<CryptoCurrency>>;
  cryptoCurrencyHoldings?: Maybe<Array<CryptoCurrencyHolding>>;
};


export type QueryCryptoCurrenciesArgs = {
  query?: Maybe<Scalars['String']>;
};
