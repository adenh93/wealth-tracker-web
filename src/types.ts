export interface Asset {
  id: string
  logoSrc: string
  name: string
  ticker: string
  price: number
  gain24Hr: number
  holdings: number
}
