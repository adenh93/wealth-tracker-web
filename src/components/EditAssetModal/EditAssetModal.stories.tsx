import { Story } from '@storybook/react'
import EditAssetModal, { EditAssetModalProps } from './EditAssetModal'
import { Asset } from '../../types'

const config = {
  title: 'Components/EditAssetModal',
  component: EditAssetModal,
}

const Template: Story<EditAssetModalProps> = (args) => (
  <EditAssetModal {...args} />
)

export const Primary = Template.bind({})

const asset: Asset = {
  id: 'test-id',
  logoSrc: 'https://upload.wikimedia.org/wikipedia/commons/4/4e/BitConnect.png',
  name: 'Bitconnect',
  ticker: 'BCC',
  price: 0.003021,
  gain24Hr: -0.9984,
  holdings: 2450000.15,
}

Primary.args = {
  asset,
  open: true,
  handleSave: () => {},
  handleClose: () => {},
}

export default config
