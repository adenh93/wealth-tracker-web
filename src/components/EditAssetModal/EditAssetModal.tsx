import { TextField, Typography } from '@mui/material'
import { ChangeEvent, FC, useState } from 'react'
import { Asset } from '../../types'
import ConfirmModal from '../ui/ConfirmModal'

export interface EditAssetModalProps {
  asset: Asset
  open: boolean
  handleSave: () => void
  handleClose: (e: any) => void
}

const EditAssetModal: FC<EditAssetModalProps> = ({
  asset,
  open,
  handleSave,
  handleClose,
}) => {
  const [newHoldings, setNewHoldings] = useState<number>(asset.holdings)

  const onChangeHoldings = (e: ChangeEvent<HTMLInputElement>) =>
    setNewHoldings(+e.currentTarget.value)

  return (
    <ConfirmModal
      open={open}
      title="Edit Asset"
      handleConfirm={handleSave}
      handleClose={handleClose}
    >
      <Typography variant="body2" sx={{ mb: 2 }}>
        Editing holdings for {asset.name} ({asset.ticker}):
      </Typography>
      <TextField
        variant="standard"
        label="New Holdings"
        type="number"
        value={newHoldings}
        onChange={onChangeHoldings}
        data-testid="new-holdings-input"
      />
    </ConfirmModal>
  )
}

export default EditAssetModal
