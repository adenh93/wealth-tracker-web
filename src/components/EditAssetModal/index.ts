export { default } from './EditAssetModal'
export type { EditAssetModalProps } from './EditAssetModal'