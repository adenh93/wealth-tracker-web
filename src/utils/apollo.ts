import { ApolloError } from '@apollo/client'

export const extractInputErrorField = <FormType>(
  error: ApolloError
): keyof FormType | null => {
  if (error.graphQLErrors.length)
    return error.graphQLErrors[0].extensions?.field as keyof FormType
  return null
}
