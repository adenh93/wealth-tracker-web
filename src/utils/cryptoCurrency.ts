import { CMC_LOGO_URL } from '../constants'

export const getCMCLogoUrl = (id: number): string => `${CMC_LOGO_URL}/${id}.png`
