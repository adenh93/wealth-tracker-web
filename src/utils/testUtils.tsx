import { ReactElement } from 'react'
import { ThemeProvider } from '@mui/material'
import { render as tlRender, RenderResult } from '@testing-library/react'
import theme from '../theme'

export const render = (component: ReactElement): RenderResult =>
  tlRender(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
