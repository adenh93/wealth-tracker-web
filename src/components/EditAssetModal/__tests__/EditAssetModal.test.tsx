import { Asset } from '../../../types'
import { render } from '../../../utils/testUtils'
import EditAssetModal from '../EditAssetModal'

describe('EditAssetModal component tests', () => {
  const mockAsset: Asset = {
    id: 'test-id',
    logoSrc:
      'https://upload.wikimedia.org/wikipedia/commons/4/4e/BitConnect.png',
    name: 'Bitconnect',
    ticker: 'BCC',
    price: 0.003021,
    gain24Hr: -0.9984,
    holdings: 2450000.15,
  }

  test('matches snapshot', () => {
    const { asFragment } = render(
      <EditAssetModal
        asset={mockAsset}
        open={true}
        handleSave={jest.fn()}
        handleClose={jest.fn()}
      />
    )

    expect(asFragment).toMatchSnapshot()
  })

  test('renders asset name correctly', () => {
    const { getByText } = render(
      <EditAssetModal
        asset={mockAsset}
        open={true}
        handleSave={jest.fn()}
        handleClose={jest.fn()}
      />
    )

    expect(getByText('Editing holdings for Bitconnect (BCC):')).not.toBeNull()
  })
})
