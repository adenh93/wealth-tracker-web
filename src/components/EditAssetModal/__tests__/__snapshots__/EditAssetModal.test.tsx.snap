// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditAssetModal component tests matches snapshot 1`] = `[Function]`;
